### PR TITLE
REF/TYP: define NDFrame numeric methods non-dynamically

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

cc @simonjayhawkins along with #36989 this gets most of the generic.py mypy complaints (notwithstanding complaints about `cls.foo = foo` that can be addressed by changing to `setattr(cls, "foo", foo)`.

it'd be nice to have a less-verbose way of pinning the docstrings.  i haven't had any luck so far at making that work.

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
> it'd be nice to have a less-verbose way of pinning the docstrings. i haven't had any luck so far at making that work.

i've spent more time than i'd like to admit looking into this.

tried altering the doc decorator to allow templates but it turns out easier to pass the actually docstring to the decorator instead of the function.

https://github.com/jbrockmendel/pandas/compare/typ-generic-3...simonjayhawkins:docs?expand=1 for a potential solution to any and all. (i've merged master so the diff is not so clear) that sits on top of this PR.

of course, now that the methods are in both DataFrame and Series, don't need the common examples.

**Comment by jbrockmendel** [comment]
> https://github.com/jbrockmendel/pandas/compare/typ-generic-3...simonjayhawkins:docs?expand=1 for a potential solution to any and all. (i've merged master so the diff is not so clear) that sits on top of this PR.

thats a nice touch putting the shared bits into `_logical_func.__doc__`

Did the `axis_descr` get lost somewhere though?

**Comment by simonjayhawkins** [comment]
> Did the `axis_descr` get lost somewhere though?

not used in any/all, `.format` does not raise KeyError with extra parameters, only when missing.

File: pandas/core/generic.py [hunk 1]
```diff
@@ -10399,37 +10398,323 @@ def _agg_by_level(self, name, axis=0, level=0, skipna=True, **kwargs):
...
     def _add_numeric_operations(cls):
         """
         Add the operations to the cls; evaluate the doc strings again
         """
         axis_descr, name1, name2 = _doc_parms(cls)
 
-        cls.any = _make_logical_function(
-            cls,
-            "any",
+        @doc(
```
Comment (jreback) [inline pandas/core/generic.py:10689]: is the point of leaving these here, to make the doc strings work / make mypy happy? e.g. you defined them above an now you are re-defining them for Series/DataFrame

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (jreback) [reply pandas/core/generic.py:10689]: > it'd be nice to have a less-verbose way of pinning the docstrings. i haven't had any luck so far at making that work.

i guess this comment explains?

Comment (jbrockmendel) [reply pandas/core/generic.py:10689]: redefining to get the docstrings right.  im not wild about this either

Comment (jbrockmendel) [reply pandas/core/generic.py:10689]: yah.  im going to try @simonjayhawkins's suggestion, so let's stick a pin in this, ill gentle-ping when ready

Comment (simonjayhawkins) [reply pandas/core/generic.py:10689]: that was my initial reaction. but am now more comfortable that this could be dealt with in a follow-on

Comment (jreback) [reply pandas/core/generic.py:10689]: either way, lgtm.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
merging to unblock a doc-issue, pls followup as needed

**Pull request merged**